pyaccel/lattice.py

```diff
@@ -559,15 +541,14 @@ def set_error_rotation_roll(lattice, indices, values):
     """
 
     # processes arguments
-    indices, values = _process_args_errors(indices, values)
+    indices, values, _ = _process_args_errors(indices, values)
 
     # loops over elements and sets its R1 and R2 fields
-    for segs, val in zip(indices, values):
-        cos, sin = _math.cos(val), _math.sin(val)
-        rot = _numpy.diag([cos, cos, cos, cos, 1.0, 1.0])
-        rot[0, 2], rot[1, 3], rot[2, 0], rot[3, 1] = sin, sin, -sin, -sin
-
-        for idx in segs:
+    for segs, vals in zip(indices, values):
+        for idx, val in zip(segs, vals):
+            cos, sin = _math.cos(val), _math.sin(val)
+            rot = _numpy.diag([cos, cos, cos, cos, 1.0, 1.0])
+            rot[0, 2], rot[1, 3], rot[2, 0], rot[3, 1] = sin, sin, -sin, -sin
```

Comment:
Bug here. There's no double loop in the original method!

Reply:
corrected